projects/form-builder/src/lib/utils/compile-segment.ts

```diff
@@ -55,15 +55,47 @@ export function compileSegment(
   if (segment.fields) {
 
     /**
-     * If it's an array fields aren' parsed
+     * If it's an array fields aren't parsed
      */
     if (segment.array) {
       // @ts-ignore
       fields = (segment.fields || []).map(fi => segment.array + fi);
     } else {
       fields = (segment.fields || [])
         // @ts-ignore
-        .reduce((acc: CompiledField[], key: string) => {
+        .reduce((acc: CompiledField[], keyObject: string | object) => {
+
+          let condition: any;
+          let key: string;
+
+          if (keyObject?.constructor === Object) {
+            condition = keyObject;
+
+            switch (condition?.action?.constructor) {
+              case Object: {
+                condition.action = [condition.action];
+                break;
+              }
+              case Array: {
+                break;
+              }
+              default: {
+                condition.action = [{}];
+                break;
+              }
+            }
+
+            condition.action.forEach(item => {
+              item.type = item.type || 'show';
+              item.function = item.function || '() => true';
```

Comment:
We should be doing the eval part here and automatically excluding any that don't evauluate properly.